main/src/ca/uwaterloo/flix/language/phase/Enumgen.scala

```diff
@@ -0,0 +1,353 @@
+      case Type.Float64 =>
+        val doubleToStringMethod = Constants.stringClass.getMethod("valueOf", classOf[Double])
+        method.visitMethodInsn(INVOKESTATIC, stringInternalName, doubleToStringMethod.getName,
+          asm.Type.getMethodDescriptor(doubleToStringMethod), false)
+      case _ =>
+        val objectToStringMethod = Constants.objectClass.getMethod("toString")
+        method.visitMethodInsn(INVOKEVIRTUAL, objectInternalName, objectToStringMethod.getName,
+          asm.Type.getMethodDescriptor(objectToStringMethod), false)
+    }
+
+    method.visitLdcInsn(")")
+
+
+    // We do this twice since there is 3 strings on the stack that we want to concat them together
+    for(_ <- 0 until 2) {
+      method.visitMethodInsn(INVOKEVIRTUAL, stringInternalName, stringConcatMethod.getName,
+        asm.Type.getMethodDescriptor(stringConcatMethod), false)
+    }
+
+    method.visitInsn(ARETURN)
+    method.visitMaxs(1, 10)
+    method.visitEnd()
+  }
+
+  /**
+    * Generate the `equals` method for the class.
+    * @param visitor class visitor
+    * @param name name of the class
+    * @param tpe descriptor of the `value` field
+    * @param fType Type of the `value` field
+    */
+  private def compileEqualsMethod(visitor: ClassWriter, name: List[String], tpe: String, fType: Type) = {
+    val clazz = Constants.objectClass
+    val objectEqualsMethod = clazz.getMethod("equals", clazz)
+    val neq = new Label() //label for when the object is not instanceof tag case
+
+    val method = visitor.visitMethod(ACC_PUBLIC, "equals", s"(${asm.Type.getDescriptor(clazz)})Z", null, null)
+    method.visitCode()
+    method.visitVarInsn(ALOAD, 1)
+    method.visitTypeInsn(INSTANCEOF, decorate(name))
+    method.visitJumpInsn(IFEQ, neq)
+    method.visitVarInsn(ALOAD, 0)
+    method.visitFieldInsn(GETFIELD, decorate(name), "value", tpe)
+    method.visitVarInsn(ALOAD, 1)
+    method.visitTypeInsn(CHECKCAST, decorate(name))
+    method.visitFieldInsn(GETFIELD, decorate(name), "value", tpe)
+
+    // This will pick the appropriate comparison for the type of the `value`
+    fType match {
+      case Type.Var(id, kind) =>  throw InternalCompilerException(s"Non-monomorphed type variable '$id in type '$fType'.")
+      case Type.Bool | Type.Char | Type.Int8 | Type.Int16 | Type.Int32 =>
+        method.visitJumpInsn(IF_ICMPNE, neq)
+        method.visitInsn(ICONST_1)
+      case Type.Int64 =>
+        method.visitInsn(LCMP)
+        method.visitJumpInsn(IFNE, neq)
+        method.visitInsn(ICONST_1)
+      case Type.Float32 =>
+        method.visitInsn(FCMPG)
+        method.visitJumpInsn(IFNE, neq)
+        method.visitInsn(ICONST_1)
+      case Type.Float64 =>
+        method.visitInsn(DCMPG)
+        method.visitJumpInsn(IFNE, neq)
+        method.visitInsn(ICONST_1)
+      case _ =>
+        method.visitMethodInsn(INVOKEVIRTUAL, asm.Type.getInternalName(clazz), objectEqualsMethod.getName,
```

Comment:
I think this code is correct when we generate tuples because we have to generate `equals()` but it won't work now. So perhaps, there is nothing `TODO` now or in future? :)